### PR TITLE
Refactoring jQuery and error fixing

### LINKED ISSUE #52: Hide forms that are not in use
Create new class on dropdown forms so that you can hide any that are not in use.

### LINKED ISSUE #54: Filter Style
Create a function that will display the selected activity type (water, sleep, activity), use display none to hide all others. 

- Filter based on input.value for all info dropdown

### PR DESCRIPTION
**Pull request by RayRedGoose**
@JamesRexMiller4 @Garrett-Iannuzzi 
This PR contains big refactoring of jQuery:
- divided big functions into small and dynamic ones.
- updated enter-keyup event, created one event handler for every keyup event (result on first gif);
- updated hiding events for pop-up elements (now block is closed if it  is not in focus or after clicking out of it (result on second gif).
- updated light/dark mode changing function (now charts color is changed faster).
- updated weekly step friend challenge (new view of chart on image).
- added filter function for widgets.
- deleted some unnecessary function.

For review you can start from the beginning of scripts.js file. All changes in index.html are not important and was done for setting up events.

Screen Shot 2019-10-28 at 12 05 19 PM

Tabing 

tab

Clicking

clicking

resolves #54
resolves #52

### CODE REVIEW

**Comment by RayRedGoose** [comment]
Add error / success notification to POST requests

Errors
Screen Shot 2019-10-28 at 1 24 25 PM

Success
Screen Shot 2019-10-28 at 1 24 47 PM

File: src/scripts.js [hunk 1]
```diff
@@ -62,49 +66,57 @@ Promise.all([
...
-  $('#user-sleep-by-week').text(sleep.returnMetricByWeek('hoursSlept', userSleepData));
-  $('#user-sleep-quality-by-week').text(sleep.returnMetricByWeek('sleepQuality', userSleepData));
-  $('#user-average-sleep-quality').text(sleep.returnAverage('sleepQuality', userSleepData));
-  $('#user-average-hours-slept').text(sleep.returnAverage('hoursSlept', userSleepData));
-  $('#user-current-step-count').text(activity.returnMetricByDate('numSteps', userActivityData, currentDate));
+  updateUserInfo();
+  updateWaterUserInfo();
+  updateSleepUserInfo();
+  updateActivityUserInfo();
+  updateFriendChallenge();
```
Comment (Garrett-Iannuzzi) [inline src/scripts.js]: Much cleaner version!

**Comment by JamesRexMiller4** [comment]
Very nice Raisa! Way to fix that FriendsChallengeWinner function, that one was a pain

File: src/scripts.js [hunk 2]
```diff
@@ -62,49 +66,57 @@ Promise.all([
...
+}
+
+function updateActivityUserInfo() {
+  let info = ['numSteps', 'minutesActive', 'flightsOfStairs'];
+  info.forEach(el => {
+    $(`#user-activity-day p[data-info=${el}]`).text(activity.returnMetricByDate(el, userActivityData, currentDate));
+    $(`.you-vs-world span[data-info="user-${el}"]`).text(activity.returnAverage(el, userActivityData));
+    $(`.you-vs-world span[data-info="all-${el}"]`).text(activity.returnAverage(el));
+  });
+  $('#user-activity-day p[data-info="miles"]').text(activity.returnMilesWalkedByDate(userActivityData, user));
```
Comment (JamesRexMiller4) [inline src/scripts.js]: Thank you for breaking up these JQuery DOM manipulations. Makes it much easier to read

**Pull request merged**